File: Leetcode/PlusOne.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        int up = 1;
        int l = digits.size();
        vector<int> res;
        for(int i = l - 1; i >= 0; --i){
            int g = digits[i] + up;
            up = g / 10;
            g %= 10;
            res.push_back(g);
        }
        if(up) res.push_back(1);
        reverse(res.begin(), res.end());
        return res;
    }
};

File: Leetcode/PlusOne.rs
// https://leetcode.com/problems/plus-one

struct Solution {}

impl Solution {
    pub fn plus_one(mut digits: Vec<i32>) -> Vec<i32> {
        for d in digits.iter_mut().rev() {
            if *d < 9 {
                *d += 1;

                return digits;
            }
            *d = 0;
        }

        digits.reverse();
        digits.push(1);
        digits.reverse();

        digits
    }
}
